### PR TITLE
Extract all Redis clients

### PR DESCRIPTION
**Pull request by pahaz**

### CODE REVIEW

**Review by akarjakin — APPROVED** [review]

File: packages/@core.keystone/redis.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ * @param {string} name -- name of redis client or the task purpose (we can use a different REDIS_URL for each name)
+ * @param {string} mode -- regular / subscriber redis client mode (read details: https://github.com/luin/ioredis#pubsub); you can also use it if you need a two redis client with different settings
+ * @param {string} opts -- RedisConfig to customize some client options; But please use don't use it!
+ * @return {import('ioredis')}
+ */
+function getRedisClient (name = 'default', mode = 'regular', opts = {}) {
+    redisLogger.info({ message: 'getRedisClient', name, mode })
+    if (!name) throw new Error('getRedisClient() without client name')
+    if (typeof name !== 'string') throw new Error('getRedisClient() name is not a string')
+    if (typeof mode !== 'string') throw new Error('getRedisClient() name is not a string')
```
Comment (AntonAL) [inline packages/@core.keystone/redis.js:22]:
```suggestion
    if (typeof mode !== 'string') throw new Error('getRedisClient() mode is not a string')
```

File: packages/@core.keystone/fields/AutoIncrementInteger/Implementation.js [hunk 2]
```diff
@@ -38,12 +43,12 @@ class AutoIncrementIntegerKnexFieldAdapter extends Integer.adapters.knex {
         // NOTE(pahaz): it's really hack! please don't copy this staff in a future! I'll find a better solution for that!
```
Comment (dkoviazin) [inline packages/@core.keystone/fields/AutoIncrementInteger/Implementation.js:43]: we can use serial integer type

**Review by dkoviazin — APPROVED** [review]

File: packages/@core.keystone/redis.js (hunk 1 above)
Comment (nomerdvadcatpyat) [inline packages/@core.keystone/redis.js:22]:
```suggestion
    if (typeof mode !== 'string') throw new Error('getRedisClient() mode is not a string')
```

File: packages/@core.keystone/fields/AutoIncrementInteger/Implementation.js [hunk 3]
```diff
@@ -1,7 +1,8 @@
 const { default: Redlock } = require('redlock')
 
 const { Integer } = require('@keystonejs/fields')
-const { taskQueue } = require('@core/keystone/tasks')
+
+const { getRedisClient } = require('../../redis')
```
Comment (AleX83Xpert) [inline packages/@core.keystone/fields/AutoIncrementInteger/Implementation.js:5]:
```suggestion
const { getRedisClient } = require('@core/keystone/redis')
```

**Review by AleX83Xpert — APPROVED** [review]

**Pull request merged**